site/documentation/content/api/tenant/index.md

```diff
@@ -243,14 +247,43 @@ published by devices as follows:
+     smaller than the *effective ttl*, then use the *ttl* value provided by the device as the new *effective ttl*.
    * does not contain a *ttl* header but a default property with name *ttl* is configured for the device (with the device
-      level taking precedence over the tenant level) and either *effective ttl* is `-1` or the default property's value
-      is smaller than the *effective ttl*, then use the configured default *ttl* value as the new *effective ttl*.
+     level taking precedence over the tenant level) and either *effective ttl* is `-1` or the default property's value
+     is smaller than the *effective ttl*, then use the configured default *ttl* value as the new *effective ttl*.
+1. If *effective ttl* is not `-1`, then set the downstream message's *ttl* header to its value (in milliseconds).
+
+Protocol adapters SHOULD use the *max-ttl-telemetry-qos0* and *max-ttl-telemetry-qos1* properties to determine the
+*effective time-to-live* for *telemetry* messages published by devices with QoS 0 or QoS 1 respectively as follows:
+
+1. Set *effective ttl* to `-1` (unlimited).
+1. If a *max-ttl-telemetry-qos0* value greater than *effective ttl* is set for the tenant, use that value as the
+   new *effective ttl*.
+1. If the message published by the device
+   * contains a *ttl* header and either *effective ttl* is `-1` or the *ttl* value (in seconds) provided by the device is
+     smaller than the *effective ttl*, then use the *ttl* value provided by the device as the new *effective ttl*.
+   * does not contain a *ttl* header but a default property with name *ttl-telemetry-qos0* is configured for the device
+     (with the device level taking precedence over the tenant level) and either *effective ttl* is `-1` or the default
+     property's value is smaller than the *effective ttl*, then use the configured default *ttl* value as the new *effective ttl*.
+1. If *effective ttl* is not `-1`, then set the downstream message's *ttl* header to its value (in milliseconds).
+
+Protocol adapters SHOULD use the *max-ttl-command-response* property to determine the *effective time-to-live* for
+*command response* messages published by devices as follows:
+
+1. Set *effective ttl* to `-1` (unlimited).
+1. If a *max-ttl-command-response* value greater than *effective ttl* is set for the tenant, use that value as the
+   new *effective ttl*.
+1. If the message published by the device
+   * contains a *ttl* header and either *effective ttl* is `-1` or the *ttl* value (in seconds) provided by the device is
+     smaller than the *effective ttl*, then use the *ttl* value provided by the device as the new *effective ttl*.
+   * does not contain a *ttl* header but a default property with name *ttl-command-response* is configured at the device's
+     tenant level and either *effective ttl* is `-1` or the default property's value is smaller than the *effective ttl*,
```

Comment:
How about changing the command response _ttl's_ section to include also device defaults  as in the _max-ttl-telemetry-qos0_ and _max-ttl-telemetry-qos1_  sections?
Eg. `(with the device level taking precedence over the tenant level)`

Reply:
thanks for spotting